mlrun/api/utils/background_tasks.py

```diff
@@ -6,63 +7,37 @@
 
 import fastapi
 import fastapi.concurrency
+import sqlalchemy.orm
 
 import mlrun.api.schemas
+import mlrun.api.utils.singletons.db
 import mlrun.api.utils.singletons.project_member
 import mlrun.errors
 import mlrun.utils.singleton
 from mlrun.utils import logger
 
 
-class Handler(metaclass=mlrun.utils.singleton.Singleton):
-    def __init__(self):
-        self._project_background_tasks: typing.Dict[
-            str, typing.Dict[str, mlrun.api.schemas.BackgroundTask]
-        ] = {}
-        self._background_tasks: typing.Dict[str, mlrun.api.schemas.BackgroundTask] = {}
-
-    def create_project_background_task(
-        self,
-        project: str,
-        background_tasks: fastapi.BackgroundTasks,
-        function,
-        *args,
-        **kwargs,
-    ) -> mlrun.api.schemas.BackgroundTask:
-        name = str(uuid.uuid4())
-        # sanity
-        if name in self._project_background_tasks:
-            raise RuntimeError("Background task name already exists")
-        background_task = self._generate_background_task(name, project)
-        self._project_background_tasks.setdefault(project, {})[name] = background_task
-        background_tasks.add_task(
-            self.background_task_wrapper, project, name, function, *args, **kwargs
-        )
-        return self.get_project_background_task(project, name)
-
+class AbstractBackgroundTaskHandler(abc.ABC):
+    @abc.abstractmethod
     def create_background_task(
-        self,
-        background_tasks: fastapi.BackgroundTasks,
-        function,
-        *args,
-        **kwargs,
+        self, *args, **kwargs
```

Comment:
removed abstract class, rather than `_generate_background_task` that is similar for both classes, I don't see reason to merge them together into one. I understand the statement that they are already in the same endpoint file, and that is the reason that I left them in the same file in utils, but it feels to me like they behave differently and have different endpoints and deserves separated classes.

Reply:
I'm fine with you thinking differently and going with 2 classes, but why having this  `_generate_background_task` duplication ? either make it a global function that both classes will use, or add a base class with this method

Reply:
There is actually no use for `_generate_background_task` in `ProjectBackgroundTasksHandler`. Removed.